tiledb/sm/buffer/buffer_list.h

```diff
@@ -99,6 +100,27 @@ class BufferList {
    */
   Status read_at_most(void* dest, uint64_t nbytes, uint64_t* bytes_read);
 
+  /**
+   * Seek to an offset, similar to lseek or fseek
+   *
+   * Whence options are:
+   *
+   *  SEEK_SET
+   *    The offset is set to offset bytes.
+   *
+   *  SEEK_CUR
+   *    The offset is set to its current location plus offset bytes
+   *
+   *  SEEK_END
+   *    This is not supported. Its purpose in lseek would be to set
+   *    the offset to the size of the BufferList plus offset bytes.
+   *
+   * @param offset offset to seek to
```

Comment:
Super picky, but most of the existing documentation for parameters use a complete-ish sentence as the description, so "@param offset Offset to seek to."